activerecord/lib/active_record/encryption/contexts.rb

```diff
@@ -32,7 +32,7 @@ module Contexts
         # Encryption contexts can be nested.
         def with_encryption_context(properties)
           self.custom_contexts ||= []
-          self.custom_contexts << default_context.dup
+          self.custom_contexts << context.dup
```

Comment:
Is `dup` enough, or would we need `deep_dup`? Is there a case where an inner context could change the outer context?

Reply:
Good point. Currently, all properties in `ActiveRecord::Encryption::Context` are single-level, i.e. none of them are arrays or hashes. We might use `deep_dup` to make it future-proof, though.

`ActiveRecord::Encryption::Scheme`, for example, accepts an array as a list of previous schemes.